From d4dae7d802c70b3141c1c8169e0b0bc6fa350901 Mon Sep 17 00:00:00 2001
From: Ruslan Bay <ruslanbey@proton.me>
Date: Thu, 26 Feb 2026 17:21:21 +0100
Subject: [PATCH 25/54] media: intel/ipu4: Update from_timer() to
 timer_container_of()

Replace from_timer() macro with timer_container_of(), which is the new kernel API
for extracting container structures from timer callbacks.
---
 drivers/media/pci/intel/ipu4/ipu4-psys.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/media/pci/intel/ipu4/ipu4-psys.c b/drivers/media/pci/intel/ipu4/ipu4-psys.c
index 34a59e84c496..74309962b152 100644
--- a/drivers/media/pci/intel/ipu4/ipu4-psys.c
+++ b/drivers/media/pci/intel/ipu4/ipu4-psys.c
@@ -744,7 +744,7 @@ void ipu_psys_watchdog_work(struct work_struct *work)
 
 static void ipu_psys_watchdog(struct timer_list *t)
 {
-	struct ipu_psys_kcmd *kcmd = from_timer(kcmd, t, watchdog);
+	struct ipu_psys_kcmd *kcmd = timer_container_of(kcmd, t, watchdog);
 	struct ipu_psys *psys = kcmd->fh->psys;
 
 	queue_work(IPU_PSYS_WORK_QUEUE, &psys->watchdog_work);
-- 
2.51.0

